src/reducers/existingThoughtDelete.js

```diff
@@ -33,6 +34,11 @@ export default (state, { context, thoughtRanked, showContexts }) => {
   const thoughtIndexNew = { ...state.thoughtIndex }
   const oldRankedThoughts = rankThoughtsFirstMatch(state, thoughts)
 
+  const isValidThought = thought.contexts.find(parent => equalArrays(context, parent.context) && rank === parent.rank)
+
+  // if thought is not valid then just return current state
+  if (!isValidThought) return { ...state }
```

Comment:
You can return nothing, as the main reducer handles it when [merging the updated fields into state](https://github.com/cybersemics/em/blob/dev/src/reducers/index.js#L98).

Can you add a `console.error` here? This *shouldn't* happen normally, so I'd like to be aware of it if it does.

Reply:
Sure!